Modified article "How to: Sort a GridView Column When a Header Is Clicked"

Gimly opened the pull request:
# Modified article "How to: Sort a GridView Column When a Header Is Clicked"

## Summary

The way the article's code example was written before this commit, the sorting was working only if the column's header text was exactly the same as the bound properties used to display, which is rarely the case in real world use.

I changed the code so that it gets the `DisplayMemberBinding` of the column and, if it is not present, will fall back to the previous implementation by getting the column header's text.

Gimly committed:
Modified article "How to: Sort a GridView Column When a Header Is Clicked"

The way the article's code example was written before this commit, the sorting was working only if the column's header text was exactly the same as the bound properties used to display, which is rarely the case in real world use.

I changed the code so that it gets the `DisplayMemberBinding` of the column and, if it is not present, will fall back to the previous implementation by getting the column header's text.

BillWagner reviewed (commented):
Thanks for updating this article @Gimly We really appreciate it.

As we find issues with samples, we're also trying to use updated language features where appropriate.  See my one suggestion. 

If you agree with the change, just push the update to your branch.

Gimly commented:
@BillWagner Sure, it was actually something that I wondered while I was making the change, whether I should use `var` and other updated language features. I'm going to rework on that.

By the way, I can't see your suggestion, could you tell me where to look, I must be missing something on GitHub's interface. Thanks

Gimly committed:
Improved code with updated language features

Improved the code of the example by using implicitly typed variable where it made sense and combining null-conditional and null coalescing operators to remove an if else.

Also improved code formatting in the VB example.

Gimly committed:
Fixed indentation in VB.Net `Sort` example

BillWagner reviewed (approved):
Thanks for making the updates @Gimly (You had guessed what my suggestions were going to be.  😄 

I must have added a comment instead of submitting a review earlier. I'm not seeing my earlier comment either.  Normally, it would show up in the *Files Changed* tab of your PR. 

At any rate, this is good, and I'll :shipit: now.  It should show up on the live site in the next day or two, on our regular publishing cycle.  

Thanks again for spotting and fixing this.  We really appreciate it

Gimly commented:
Thanks for the kind words, glad to have been of help.
